src/staticanalysis/bot/static_analysis_bot/workflows/remote.py

```diff
@@ -0,0 +1,132 @@
+            if artifact['storageType'] != 'reference' and artifact['contentType'].startswith('text/')
+        ]
+
+        # Load logs from artifact API
+        out = {}
+        for log in logs:
+            logger.info('Reading log', task_id=self.task_id, log=log)
+            try:
+                artifact = queue_service.getArtifact(self.task_id, self.run_id, log)
+                assert 'response' in artifact, 'Failed loading artifact'
+                out[log] = artifact['response'].content
+            except Exception as e:
+                logger.warn('Failed to read log', task_id=self.task_id, run_id=self.run_id, log=log, error=e)
+                raise
+        return out
+
+
+class RemoteWorkflow(object):
+    '''
+    Secondary workflow to analyze the output from a try task group
+    '''
+    def __init__(self, queue_service):
+        # Use TC services client
+        self.queue_service = queue_service
+
+    def run(self, revision):
+        assert settings.taskcluster.task_id is not None, 'Missing Taskcluster task id'
+
+        # Load task description
+        task = self.queue_service.task(settings.taskcluster.task_id)
+        assert len(task['dependencies']) > 0, 'No task dependencies to analyze'
+
+        # Find issues in dependencies
+        issues = []
+        for dep_id in task['dependencies']:
+            try:
+                task = AnalysisTask(dep_id)
+                logs = task.load_logs(self.queue_service)
+                if logs is None:
+                    continue
+
+                for log in logs.values():
+                    issues += self.parse_issues(task, log, revision)
+            except Exception as e:
+                logger.warn('Failure during task analysis', task=settings.taskcluster.task_id, error=e)
+                raise
+                continue
+
+        return issues
+
+    def parse_issues(self, task, log_content, revision):
+        '''
+        Parse issues from a log file content
+        '''
+        assert isinstance(task, AnalysisTask)
+
+        # Lookup issues using marker
+        issues = [
+            line[line.index(ISSUE_MARKER) + len(ISSUE_MARKER):]
+            for line in log_content.decode('utf-8').splitlines()
+            if ISSUE_MARKER in line
+        ]
+        assert len(issues) > 0, 'No issues found in failure log'
+
+        # Convert to Issue instances
+        logger.info('Found {} issues !'.format(len(issues)))
+        return [
+            self.build_issue(task.name, issue, revision)
+            for issue in issues
+        ]
+
+    def build_issue(self, task_name, issue, revision):
+        '''
+        Convert a raw text issue into an Issue instance
+        TODO: this should be simplified by using mach JSON output
+        '''
+        if task_name.startswith(MozLintIssue.TRY_PREFIX):
+            return MozLintIssue.from_try(task_name, issue, revision)
+        else:
+            raise Exception('Unsupported task type {}'.format(task_name))
```

Comment:
A question: when we push to try, we automatically get more than just mozlint running. So, will we always fail here?

Reply:
Fixed it